opts/opts.go

```diff
@@ -267,10 +267,24 @@ func validateDomain(val string) (string, error) {
 }
 
 // ValidateLabel validates that the specified string is a valid label, and returns it.
-// Labels are in the form on key=value.
+//
+// Labels are in the form of key=value; key must be a non-empty string, and not
+// contain whitespaces. A value is optional (defaults to an empty string if omitted).
+//
+// Leading whitespace is removed during validation but values are kept as-is
+// otherwise, so any string value is accepted for both, which includes whitespace
+// (for values) and quotes (surrounding, or embedded in key or value).
+//
+// TODO discuss if quotes (and other special characters) should be valid or invalid for keys
+// TODO discuss if leading/trailing whitespace in keys should be preserved (and valid)
 func ValidateLabel(val string) (string, error) {
-	if strings.Count(val, "=") < 1 {
-		return "", fmt.Errorf("bad attribute format: %s", val)
+	arr := strings.SplitN(val, "=", 2)
+	key := strings.TrimLeft(arr[0], whiteSpaces)
+	if key == "" {
+		return "", fmt.Errorf("empty label name: '%s'", val)
```

Comment:
Isn't the label `name` the `key`? If so you should replace `val` by `key`.

Reply:
Well ok as the key is empty, there's no point printing it in the message... Then maybe rephrase it with something like: `"invalid label '%s': empty name", val` ?

Reply:
Yes, it's not proper english; I copied these from https://github.com/docker/cli/blob/9e712073274021ef1e552ecbb54595142b49edbb/opts/file.go#L51-L54

but perhaps I should update both